pkg/eventlistener/eventlistener_test.go

```diff
@@ -0,0 +1,176 @@
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              "el2",
+				Namespace:         "ns",
+				CreationTimestamp: v1.Time{Time: clock.Now().Add(-5 * time.Minute)},
+			},
+		},
+	}
+
+	cs := test.SeedTestResources(t, triggertest.Resources{EventListeners: els, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	cs2 := test.SeedTestResources(t, triggertest.Resources{EventListeners: els2, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+	p2 := &test.Params{Triggers: cs2.Triggers, Kube: cs2.Kube}
+	p3 := &test.Params{Triggers: cs2.Triggers, Kube: cs2.Kube}
+	p3.SetNamespace("unknown")
+
+	testParams := []struct {
+		name   string
+		params *test.Params
+		want   []string
+	}{
+		{
+			name:   "Single EventListener",
+			params: p,
+			want:   []string{"el1"},
+		},
+		{
+			name:   "Multi EventListeners",
+			params: p2,
+			want:   []string{"el1", "el2"},
+		},
+		{
+			name:   "Unknown namespace",
+			params: p3,
+			want:   []string{},
+		},
+	}
+
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			got, err := GetAllEventListenerNames(tp.params.Triggers, tp.params.Namespace())
+			if err != nil {
+				t.Errorf("unexpected Error")
+			}
+			test.AssertOutput(t, tp.want, got)
+		})
+	}
+}
+
+func TestEventListener_List(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+
+	els := []*v1alpha1.EventListener{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              "el1",
+				Namespace:         "ns",
+				CreationTimestamp: v1.Time{Time: clock.Now().Add(-5 * time.Minute)},
+			},
+		},
+	}
+
+	els2 := []*v1alpha1.EventListener{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              "el1",
+				Namespace:         "ns",
+				CreationTimestamp: v1.Time{Time: clock.Now().Add(-5 * time.Minute)},
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              "el2",
+				Namespace:         "ns",
+				CreationTimestamp: v1.Time{Time: clock.Now().Add(-5 * time.Minute)},
+			},
+		},
+	}
+
+	cs := test.SeedTestResources(t, triggertest.Resources{EventListeners: els, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	cs2 := test.SeedTestResources(t, triggertest.Resources{EventListeners: els2, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+	p2 := &test.Params{Triggers: cs2.Triggers, Kube: cs2.Kube}
+
+	testParams := []struct {
+		name   string
+		params *test.Params
+		want   []string
+	}{
+		{
+			name:   "Single EventListener",
+			params: p,
+			want:   []string{"el1"},
+		},
+		{
+			name:   "Multi EventListeners",
+			params: p2,
+			want:   []string{"el1", "el2"},
+		},
+	}
+
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			got, err := GetAllEventListenerNames(tp.params.Triggers, "ns")
```

Comment:
I guess it's checking for names here. Could you please change it to `List`

Reply:
nice catch :+1: